third-party/realdds/src/dds-option.cpp

```diff
@@ -0,0 +1,42 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+#include <realdds/dds-option.h>
+#include <realdds/dds-exceptions.h>
+
+#include <librealsense2/utilities/json.h>
+using nlohmann::json;
+
+
+namespace realdds {
+
+
+dds_option::dds_option( nlohmann::json const & j )
+{
+    int index = 0;
+    _name = utilities::json::get< std::string >( j, index++ );
+    _value = utilities::json::get< float >( j, index++ );
+    _range.min = utilities::json::get< float >( j, index++ );
+    _range.max = utilities::json::get< float >( j, index++ );
+    _range.step = utilities::json::get< float >( j, index++ );
+    _range.default_value = utilities::json::get< float >( j, index++ );
+    _description = utilities::json::get< std::string >( j, index++ );
+
+    if( index != j.size() )
+        DDS_THROW( runtime_error, "expected end of json at index " + std::to_string( index ) );
+}
+
+
+/* static  */ std::shared_ptr< dds_option > dds_option::from_json(nlohmann::json const & j)
+{
+    return std::shared_ptr< dds_option >( new dds_option( j ) );
+}
+
+
+nlohmann::json dds_option::to_json() const
+{
+    // Note - same order as construction!
+    return json::array( { _name, _value, _range.min, _range.max, _range.step, _range.default_value, _description } );
```

Comment:
I think, in the case of options, we want a dictionary and not an array. We can see what the result size is, but let's start with the flexible approach and then see how we can optimize.

Reply:
I will consider changing to a separate message as mentioned in another comment, but in this implementation as part of the large stream-header message we should conserve the place